web/src/screens/private/Events/EditEvent.js

```diff
@@ -4,31 +4,159 @@ import { compose } from 'recompose';
+      name: locationToAdd.name,
+      detail: locationToAdd.detail,
+      icon: locationToAdd.icon,
+      locationLatitude: locationToAdd.locationLatitude,
+      locationLongitude: locationToAdd.locationLongitude,
+    });
+  }
+
+  return locations;
+}
+
 class EditEvent extends React.Component {
   state = {
+    id: 0,
+    groupId: '',
+    groupName: '',
     name: '',
     details: '',
+    activeStep: 0,
+    scenesList: defaultScenes,
+    lhqsList: defaultLhqs,
+    helosList: defaultHelos,
+  };
+
+  componentWillReceiveProps(nextProps) {
+    if (
+      nextProps.loading === false &&
+      nextProps.match.params.id !== undefined &&
+      this.state.id === 0
+    ) {
+      const event = nextProps.user.events.find(
+        e => e.id === parseInt(this.props.match.params.id, 10),
+      );
+
+      if (event !== undefined) {
+        let scene = event.eventLocations.find(s => s.icon === 'scene');
+        let lhq = event.eventLocations.find(s => s.icon === 'lhq');
+        let helo = event.eventLocations.find(s => s.icon === 'helo');
+
+        const scenes = [...defaultScenes];
+        const lhqs = [...defaultLhqs];
+        const helos = [...defaultHelos];
+
+        if (scene !== undefined) {
+          const loc = scenes.find(
+            newLoc =>
+              newLoc.locationLatitude === scene.locationLatitude &&
+              newLoc.locationLongitude === scene.locationLongitude,
+          );
+          if (loc === undefined) {
+            scenes.push(scene);
+          } else {
+            scene = loc;
+          }
+        }
+
+        if (lhq !== undefined) {
+          const loc = lhqs.find(
+            newLoc =>
+              newLoc.locationLatitude === lhq.locationLatitude &&
+              newLoc.locationLongitude === lhq.locationLongitude,
+          );
+
+          if (loc === undefined) {
+            lhqs.push(lhq);
+          } else {
+            lhq = loc;
+          }
+        }
+
+        if (helo !== undefined) {
+          const loc = helos.find(
+            newLoc =>
+              newLoc.locationLatitude === helo.locationLatitude &&
+              newLoc.locationLongitude === helo.locationLongitude,
+          );
+
+          if (loc === undefined) {
+            helos.push(helo);
+          } else {
+            helo = loc;
+          }
+        }
+
+        this.setState({
+          id: event.id,
+          name: event.name,
+          details: event.details,
+          groupId: event.group.id,
+          groupName: event.group.name,
+          scene,
+          lhq,
+          helo,
+          scenesList: scenes,
+          lhqsList: lhqs,
+          helosList: helos,
+        });
+      }
+    }
+  }
+
+  onGroupChange = (e) => {
+    const groupId = e.target.value;
+    let name = '';
+
+    const group = this.props.user.groups.find(g => g.id === parseInt(groupId, 10));
+    if (group !== undefined) ({ name } = group);
+
+    this.setState(() => ({ groupId, groupName: name }));
```

Comment:
any reason you use the callback form of setState() instead of passing a dict directly? might be good to capture this in comments

Reply:
Oversight - I'll correct.